service/src/main/java/com/epam/esm/service/impl/GiftCertificateServiceImpl.java

```diff
@@ -0,0 +1,240 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.dao.GiftCertificateDAO;
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dao.exception.PersistenceException;
+import com.epam.esm.model.GiftCertificate;
+import com.epam.esm.model.Tag;
+import com.epam.esm.service.request.CertificateRequestBody;
+import com.epam.esm.dao.exception.ErrorCodeEnum;
+import com.epam.esm.service.GiftCertificateService;
+import com.epam.esm.service.exception.ServiceException;
+import com.epam.esm.service.request.SortParameter;
+import com.epam.esm.service.request.SortType;
+import com.epam.esm.service.util.CertificateValidator;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+public class GiftCertificateServiceImpl implements GiftCertificateService {
+
+    private static final Logger LOGGER = LogManager.getLogger(GiftCertificateServiceImpl.class);
+
+    private final GiftCertificateDAO giftCertificateDAO;
+    private final TagDao tagDao;
+    private final CertificateValidator certificateValidator;
+
+    @Autowired
+    public GiftCertificateServiceImpl(GiftCertificateDAO giftCertificateDAO, TagDao tagDao,
+                                      CertificateValidator certificateValidator) {
+        this.giftCertificateDAO = giftCertificateDAO;
+        this.tagDao = tagDao;
+        this.certificateValidator = certificateValidator;
+    }
+
+    @Override
+    public GiftCertificate getGiftCertificate(String name) throws ServiceException {
+        certificateValidator.validateName(name);
+        try {
+            return giftCertificateDAO.getGiftCertificate(name);
+        } catch (DataAccessException e) {
```

Comment:
Do you really need try/catch here?
If certificate not found would dao return null?